src/global/global.cpp

```diff
@@ -452,6 +409,65 @@ void Init_Param_Struct_Members(ParameterMap &pmap, struct Parameters *parms)
   parms->gamma = Real(pmap.value<double>("gamma"));
   CHOLLA_ASSERT(parms->gamma > 1.0, "gamma parameter must be greater than one.");
 
+  // load in a handful of string parameters (this would look a lot more like parsing other parameters if we
+  // stored the values as std::string values)
+  Load_String_Param_Into_Char_Buffer(pmap, "init", parms->init, "");
+  Load_String_Param_Into_Char_Buffer(pmap, "custom_bcnd", parms->custom_bcnd, "");
+  Load_String_Param_Into_Char_Buffer(pmap, "outdir", parms->outdir, "");
+  Load_String_Param_Into_Char_Buffer(pmap, "indir", parms->indir, "");
+
+  // in the future, the feedback module will read in its own parameters (the global Parameter struct won't
+  // know anything about it)
+#ifdef FEEDBACK
+  #ifndef NO_SN_FEEDBACK
+  Load_String_Param_Into_Char_Buffer(pmap, "snr_filename", parms->snr_filename, "");
+  #endif
+  #ifndef NO_WIND_FEEDBACK
+  Load_String_Param_Into_Char_Buffer(pmap, "sw_filename", parms->sw_filename, "");
+  #endif
+#endif
+
+  // in the future, it would probably be good to move this logic into Cosmology::Initialize (or somewhere similar)
+  // and remove these parameters from the global struct. This would provide a few benefits:
+  //   - we could take steps towards removing the optionalParams global variable (there is alternative machinery
+  //     in place to check for unused parameters)
+  //   - we could remove ifdef statements from here and the global Parameters struct (we would also remove the
+  //     these parameters from the Parameters struct)
+  //
+  // Prior to relocating this parameter-parsing, Cosmological simulations simply assumed that all of these parameters
+  // were specified (& there were no default values). Now cosmological simulations will loudly fail if a user forgets
+  // one of these parameters (other that wa, which defaults to 0).
+#ifdef COSMOLOGY
+  Load_String_Param_Into_Char_Buffer(pmap, "scale_outputs_file", parms->scale_outputs_file, nullptr);
+  parms->Init_redshift = pmap.value<double>("Init_redshift");
+  parms->End_redshift  = pmap.value<double>("End_redshift");
+  parms->H0            = pmap.value<double>("H0");
+  parms->Omega_M       = pmap.value<double>("Omega_M");
+  parms->Omega_L       = pmap.value<double>("Omega_L");
+  parms->Omega_b       = pmap.value<double>("Omega_b");
+  parms->Omega_R       = pmap.value<double>("Omega_R");
+  parms->w0            = pmap.value<double>("w0");
+  parms->wa            = pmap.value_or("wa", 0.0);
+#endif  // COSMOLOGY
+
+#if defined(CHEMISTRY_GPU) || defined(COOLING_GRACKLE)
+  Load_String_Param_Into_Char_Buffer(pmap, "scale_outputs_file", parms->scale_outputs_file, "");
+#endif
```

Comment:
Shouldn't `scale_outputs_file` be `UVB_rates_file` here?

Reply:
thanks for catching that!!